vue.config.js

```diff
@@ -46,6 +46,14 @@ module.exports = {
     ]
   },
   chainWebpack: config => {
+    config
+      .resolve
+      .alias
+      .set('react', 'preact/compat')
+    config
+      .resolve
+      .alias
+      .set('react-dom', 'preact/compat')
```

Comment:
This way Webpack will provide the link for the applications using React, and they will get preact instead.